From 69baca5c499ba48c5a6deade50b57a7c097f5c3c Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 25 May 2022 06:08:53 +0300
Subject: [PATCH] [SW-0] rom: add smi control logic

As per the updated security specification, smi enable/disable logic
is added in rom.

Change-Id: I40efede9cd071b4de88cb32884eb5c2aad3195f4
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 00ee80e..729beb2 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -505,12 +505,21 @@
 	regmap_writel(AXPROT_SEC_VAL, mmD0_SMI_MSTR_IF_AXPROT_HBW_ARPROT_1_OVRD);
 }
 
+#define SMI_EN_VAL		1
+#define SMI_DIS_VAL		0
+void security_smi_access(uint8_t val)
+{
+	regmap_writel(val, mmD0_PSOC_GLOBAL_CONF_SMI_ACCESS_EN);
+}
+
 void security_enable_twi_if(void)
 {
+	security_smi_access(SMI_EN_VAL);
 }
 
 void security_disable_twi_if(void)
 {
+	security_smi_access(SMI_DIS_VAL);
 	security_i2c_axprot_config();
 	security_smi_axprot_config();
 }
